### ISSUE TITLE
Checkout Promo Code - UX Enhancements

### ISSUE THREAD

**Issue by brittanygay** [revelrylabs/bn-mobile-react#563]
_Broken out from #524_ 

Per our discussion in https://github.com/revelrylabs/bn-mobile-react/pull/554, we could enhance this layout to have a better the UX. 

It looks like I should click on the 'review promo code' button moreso than anything else on this screen. Im not sure what a user should do once they've added a successful promo code. (Click the back button? Click on the ticket row? Close the app?)

Everyone on the team agrees with the above experience, but more suggestions are welcome in this ticket.

**Comment by brittanygay** [comment]
Assigning to @Mcscott142 to approve the above AC

**Comment by daybreaker** [comment]
One idea is showing the original price crossed out maybe, with the new price, and an extra row on the ticket saying `Promo applied` or something?

**Comment by zmelamed** [comment]
@brittanygay any more thoughts on making it clearer that the user needs to select the ticket type to proceed?

**Comment by brittanygay** [comment]
Gathering some thoughts for this now, will post what I have soon

**Comment by brittanygay** [comment]
**Here's a detailed solution with some quick wires reflecting my ideas for the promo code UX:**

1. Move the promo code section to the _next_ screen, so it will be above the "Enter Paymnent" section. This keeps all payment options in one place, and it makes it easier for the user to understand

![563-1](https://user-images.githubusercontent.com/14582709/54624477-6b958e00-4a3b-11e9-8e19-f3419f81ded9.JPG)

![563-2](https://user-images.githubusercontent.com/14582709/54624478-6b958e00-4a3b-11e9-8812-a80dd41f45ba.JPG)


2. The "Promo Code" section will be a screen by itself, just the same as the payment information screen is set up.

![563-3](https://user-images.githubusercontent.com/14582709/54624510-7d773100-4a3b-11e9-83d6-b163c7bd7e28.JPG)


3. Once the promo code is added, the user is then directed to continue to complete the checkout flow

![563-4](https://user-images.githubusercontent.com/14582709/54624569-a4356780-4a3b-11e9-9089-642c02c13b0c.JPG)


4. Once a promo code is added, it will reflect on the subtotal section, showing the user how much the promo code deducts the total price

![563-5](https://user-images.githubusercontent.com/14582709/54624622-badbbe80-4a3b-11e9-93d8-12c86470c2d1.jpg)


- Overall, this will also make the userflow easier for the user to understand, showing only one option to move forward with the checkout flow, and adding a promo code if they desire
- Showing the promo code added to the total will also help clear up confusion once they are ready to submit their payment (I'm not sure if this was here before, I can't test the promo codes since I don't have any test codes to use)

Assigning over to @zmelamed to take a look!

**Comment by zmelamed** [comment]
This is a quick concept, but what about something like this:
![image](https://user-images.githubusercontent.com/456953/54643838-16667600-4a55-11e9-997b-9d1e9d81dba9.png)
The issue is:
1) We want to implement something that isn't significantly different from what is there because we're likely going to do a bigger overhaul of checkout to simplify
2) Promo code needs to be part of the first step because there are promo codes that actually expose tickets that aren't yet visible, so it can't be applied after the tickets are selected.

Can you iterate on/clean up the above approach where we wouldn't change too much on that view

**Comment by Mcscott142** [comment]
@brittanygay see above comment.

**Comment by brittanygay** [comment]
Ah I see what you mean @zmelamed, I wasn't aware that we would be changing up the checkout flow so much and how the promo code reveals new tickets. 

Let me sketch up some new wire solutions and I'll post in here today

**Comment by brittanygay** [comment]
@zmelamed after reviewing your notes a little more, I like what you have mocked up and I agree on this new approach. Here's my reasoning:

1. The "Apply Code" now looks more like it's an action within the Promo section only, instead of a button sitting underneath that may confuse the user/make them want to click the Remove button rather than move onto the rest of the Checkout userflow
2. "Remove Code" can look the same once a code is applied, and we could use either red text or have an X icon next to it to help indicate so. 
3. It is clear that the Promo Code section is separate from the Ticket Type section, plus the user can clearly see what new tickets are exposed once the promo code is applied
4. It's also clear that the user needs to then select a ticket type in order to continue checking out

I do think it would be nice to keep any promo code price deductions listed under the Subtotal on the Checkout flow (if there are any), so it's clear to the user that their promo code was applied when they are about to purchase their ticket.

**Comment by brittanygay** [comment]
Assigning back to @zmelamed so we can come to a conclusion and get this rolling!

**Comment by zmelamed** [comment]
Yeah lets just keep it without the promo code price change as I don't know that that is actually available on the client side and just for simplicity sake for now.

One thing to also keep in mind when implementing this is that the event descriptions can be very long, or they may be non existent. So when you implement this please keep that in mind, the price and the select button should remain at the top with the ticket type name rather than floating to the vertical middle of the ticket type cart if the description becomes lengthy.

Also, the select button should probably just switch to a grey "sold out" state if it is sold out

**Comment by iamjoshfrank** [comment]
Removing `SCORE` label, changing to `draft`. Based on the Issue comments, it looks like we need to write a clear Scenario and Acceptance Criteria upon which we can implement the solution.
